## Generate BERT Embeddings

# Import Model

In [1]:
from transformers import BertModel, BertTokenizer
import torch

# Load tokenizer and model

In [4]:
tokenizer= BertTokenizer.from_pretrained('bert-base-uncased')
model= BertModel.from_pretrained('bert-base-uncased')

C:\Users\Aman Sinha\AppData\Local\Programs\Python\Python311\Lib\site-packages\huggingface_hub\file_download.py:133: UserWarning: `huggingface_hub` cache-system uses symlinks by default to efficiently store duplicated files but your machine does not support them in C:\Users\Aman Sinha\.cache\huggingface\hub. Caching files will still work but in a degraded version that might require more space on your disk. This warning can be disabled by setting the `HF_HUB_DISABLE_SYMLINKS_WARNING` environment variable. For more details, see https://huggingface.co/docs/huggingface_hub/how-to-cache#limitations.
To support symlinks on Windows, you either need to activate Developer Mode or to run Python as an administrator. In order to see activate developer mode, see this article: https://docs.microsoft.com/en-us/windows/apps/get-started/enable-your-device-for-development
  warnings.warn(message)


# Input text

In [6]:
text = "I want to kill myself again"
inputs = tokenizer(text, return_tensors='pt')


## Get Embeddings

In [7]:
outputs = model(**inputs)
embeddings = outputs.last_hidden_state

In [9]:
print("Shape of embeddings:", embeddings.shape)

Shape of embeddings: torch.Size([1, 8, 768])


| Dimension | Meaning                                                |
| --------- | ------------------------------------------------------ |
| `1`       | Batch size                                             |
| `4`       | Number of tokens (e.g., \[CLS], I, love, transformers) |
| `768`     | Each token represented by a 768-dimensional vector     |
